TTHAnalysis/macros/prepareEventVariablesFriendTree.py

```diff
@@ -266,7 +268,8 @@ def analyze(self,event):
             fname    = open(fname+".url","r").readline().strip()
     if os.path.exists(fname) or (os.path.exists("%s/%s/tree.root.url" % (D,options.tree))):
         short = os.path.basename(D)
-        if short.endswith(".root"): short = short[:-len(".root")] # rstrip does not do what one would like
+        if not isNano: 
+            if short.endswith(".root"): short = short[:-len(".root")] # rstrip does not do what one would like
```

Comment:
This change introducing `is not Nano:` seems wrong to me.
* For NanoAOD, it can happen that `D` is a rootfile (see line 253-254). As such, it's correct to strip the `.root` in order to derive the sample name.
* For non-nanoAOD cases, `D` should normally be a directory, and so it shouldn't end in `.root` (unless you're intentionally trying to confuse people), so the line should be harmless

Reply:
From what I see by running, for NanoAOD when =D= is a rootfile then stripping the .root makes the dataset name match fail.
I concluded that if =D= is a rootfile the =.root= suffix should not be removed, and that the line was actually a protection for non-nanoAOD against the case in which =D= is incorrectly a file.

If I remove the "if not isNano", the script does not find any match and does not run anything, in concrete.

Reply:
are you including ".root" in the dataset name?
the code is assuming that the dataset names do not contain .root, just like in the mca files.

Reply:
The current version of the script seems to be including ".root" in the dataset name; I did not configure it as such, it came like that with the new branch.
Perhaps this is a consequence of the overhauling of the script by @sscruz?

Reply:
hi, I don't see how my PR can change the behaviour of that... I'm not including the .root in the dataset name